pkg/payload/task_graph.go

```diff
@@ -514,9 +514,8 @@ func RunGraph(ctx context.Context, graph *TaskGraph, maxParallelism int, fn func
 				inflight--
 			case <-ctx.Done():
 				select {
-				case runTask := <-workCh: // workers canceled, so remove any work from the queue ourselves
+				case <-workCh: // workers canceled, so remove any work from the queue ourselves
```

Comment:
NIT:

```suggestion
				case <-workCh: // Reset the inflight counter as workers got canceled. We never submit more work once canceled and thus no need to reset the submitted records.
```

Reply:
The important part is still the "remove any work from the queue" though - that needs to stay. The channel gets drained (eventually...).

I'm not convinced about the usefulness of mentioning the submitted records - it explains the absence of code, but that only makes sense attached to a commit/code change, does not make that much sense in the actual code? "We are _not_ doing something because it _does not_ need to be done"-like comment IMO makes sense only if that absence is actually surprising, which does not seem to be the case here?

Reply:
> the "remove any work from the queue" though - that needs to stay.

I think i got it now.
"remove any work from the queue ourselves" corresponds to `<-workCh` (that is why they stay in the same line). 

My mistake was I thought the comment was for the code below,  currently having only `inflight--` which i couldn't relate to the comment. Actually, `inflight--` is the additional things we need to do when `<-workCh` takes place.

> We are not doing something because it does not need to be done

I got the point.

> like comment IMO makes sense only if that absence is actually surprising,

It would probably surprise the author of "submitted[runTask.index] = false". 😉  When we push a task, we do `submitted[nextNode] = true`, my intuitive feeling is that we would set it to false when it is removed from the queue before the task is completed. With a comment, it would be telling that we do _not_ do it intentionally, instead of forgetting it. Maybe it is just me. Just a NIT anyway.

Reply:
> It would probably surprise the author of "submitted[runTask.index] = false"

:grin:

Reply:
I'm the author of that line, in 632e763fa209 (#455), and Petr's "We never submit more work once canceled..." makes sense to me, and if I `git blame ...` the line with the comment, I'll find his commit message, so all good on that side :)